src/shadowbox/integration_test/test.sh

```diff
@@ -71,18 +79,64 @@ function ss_arguments_for_user() {
 
 # Runs curl on the client container.
 function client_curl() {
-  docker exec "${CLIENT_CONTAINER}" curl --silent --show-error --connect-timeout 5 --retry 5 "$@"
+  podman exec "${CLIENT_CONTAINER}" curl --silent --show-error --connect-timeout 5 --retry 5 "$@"
 }
 
 function fail() {
   echo FAILED: "$@"
   exit 1
 }
 
+function setup() {
+  shutdown_containers
+
+  podman network create -d bridge "${NET_OPEN}"
+  podman network create -d bridge --internal "${NET_BLOCKED}"
+
+  # Target service.
+  podman build --force-rm -t "${TARGET_IMAGE}" ./target
+  podman run -d --rm -p "10080:80" --network="${NET_OPEN}" --network-alias="target" --name="${TARGET_CONTAINER}" "${TARGET_IMAGE}"
+  
+  # Shadowsocks service.
+  declare -ar shadowbox_flags=(
+    -d
+    --rm
+    --network="${NET_BLOCKED}"
+    --network-alias="shadowbox"
+    -p "20443:443"
+    -e "SB_API_PORT=443"
+    -e "SB_API_PREFIX=${SB_API_PREFIX}"
+    -e "LOG_LEVEL=debug"
+    -e "SB_CERTIFICATE_FILE=/root/shadowbox/test.crt"
+    -e "SB_PRIVATE_KEY_FILE=/root/shadowbox/test.key"
+    -v "${SB_CERTIFICATE_FILE}:/root/shadowbox/test.crt"
+    -v "${SB_PRIVATE_KEY_FILE}:/root/shadowbox/test.key"
+    -v "${TMP_STATE_DIR}:/root/shadowbox/persisted-state"
+    --name "${SHADOWBOX_CONTAINER}"
+    "${SB_IMAGE:-localhost/outline/shadowbox:latest}"
+  )
+  podman run "${shadowbox_flags[@]}"
+  # podman network connect --alias shadowbox "${NET_BLOCKED}" "${SHADOWBOX_CONTAINER}"
+  podman network connect "${NET_OPEN}" "${SHADOWBOX_CONTAINER}"
+
+  # Client service.
+  podman build --force-rm -t "${CLIENT_IMAGE}" ./client
+  # Use -i to keep the container running.
+  podman run -d --rm -it -p "30555:555" --network "${NET_BLOCKED}" --name "${CLIENT_CONTAINER}" "${CLIENT_IMAGE}"
+}
+
+function shutdown_containers() {
+    podman rm -f -v "${TARGET_CONTAINER}" || true
+    podman rm -f -v "${SHADOWBOX_CONTAINER}" || true
+    podman rm -f -v "${CLIENT_CONTAINER}" || true
+    podman network rm -f "${NET_OPEN}" || true
+    podman network rm -f "${NET_BLOCKED}" || true
```

Comment:
What I like to do for cleanups is add a consistent tag to all containers, images, and networks. Then you can remove everything by tag, e.g. `docker network rm -f $(docker network ls -q --filter "label=org.getoutline.tag=${TAG}")`. (I assume podman can do the same thing?) I wonder if that could help scale this? Just a thought.

Reply:
I like the idea, but the integration test should be able to test the image without affecting the image building. So no labels. That could break an existing container, for instance.
However, we can probably use the names the same way. I already have the names with a prefix

Reply:
Oh, I didn't realize we can set labels at run time. I thought it was build time only. We should consider that, but not for this PR.